internal/app/frontend/frontend_service.go

```diff
@@ -354,7 +354,62 @@ func doWatchAssignments(ctx context.Context, id string, sender func(*pb.Assignme
 // AcknowledgeBackfill is used to notify OpenMatch about GameServer connection info.
 // This triggers an assignment process.
 func (s *frontendService) AcknowledgeBackfill(ctx context.Context, req *pb.AcknowledgeBackfillRequest) (*pb.Backfill, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	if req.GetBackfillId() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, ".BackfillId is required")
+	}
+	if req.GetAssignment() == nil {
+		return nil, status.Errorf(codes.InvalidArgument, ".Assignment is required")
+	}
+
+	m := s.store.NewMutex(req.GetBackfillId())
+
+	err := m.Lock(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if _, err = m.Unlock(ctx); err != nil {
+			logger.WithError(err).Error("error on mutex unlock")
+		}
+	}()
+
+	bf, associatedTickets, err := s.store.GetBackfill(ctx, req.GetBackfillId())
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.store.AcknowledgeBackfill(ctx, req.GetBackfillId())
+	if err != nil {
+		return nil, err
+	}
+
+	if len(associatedTickets) != 0 {
+		_, tickets, err := s.store.UpdateAssignments(ctx, &pb.AssignTicketsRequest{
+			Assignments: []*pb.AssignmentGroup{&pb.AssignmentGroup{TicketIds: associatedTickets, Assignment: req.GetAssignment()}},
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		// all assigned tickets should be removed from the associatedTickets
```

Comment:
Note that `UpdateAssignment` can return not the full list and some tickets are left unassigned.

Reply:
Looking at the code in UpdateAssignment, any ticket left unassigned is because it was not found. (See: AssignmentFailure_TICKET_NOT_FOUND). So currently, any ticket that is returned as not found from UpdateAssignment is still left on the Backfill? We should atleast log the tickets that we leave unassigned here as UpdateAssignment returns a list of errors along with a list of assigned tickets (currently we ignore the errors list)

Reply:
I have added logging of the errors on assignTickets request.